### ISSUE TITLE
User documentation/guide, tutorial, FAQ

### ISSUE THREAD

**Issue by portah** [common-workflow-language/common-workflow-language#167]
We need a tutorial and user guide and a FAQ. It is a lot of writing and it has to be maintained with drafts. I believe we can achieve that and here is my penny:
https://github.com/common-workflow-language/workflows/wiki/Tutorial

I've just started but hope will keep it updated.

**Comment by pgrosu** [comment]
Hi Andrey,

This is really great and very appreciated!  

Many thanks for starting this,
Paul

**Comment by denis-yuen** [comment]
Trivial note for future me: The link above is now https://github.com/common-workflow-language/workflows/wiki/Tutorial-DRAFT2

**Comment by mikisvaz** [comment]
Hi, I've been asked to contribute some comments to the tutorial draft:

- The examples require having some non-trivial software installed like bedtools, this makes following the tutorial harder. 
- Downloading and running bedtools-genomecov.cwl did not work for me:  `Error reading file:///home/mvazquezg/git/sandbox/CWL/envvar-global.cwl [Errno 2] No such file or directory: '/home/mvazquezg/git/sandbox/CWL/envvar-global.cwl'` Removing the requirement line for that fixed it.
- The example cannot be ran without the appropriate input BAM.
- The cwl files linked for download do not match the ones described in the tutorial. For instance the linus-sort.cwl does not include an `arguments:` entry. The description on that particular part is very difficult to follow unless you know what you are trying to do. Even then, it is not clear where the `arguments:` entry should be
- The Workflow section is very a complete mystery to me, how does the tool know what CWLs to run? The script is not linked for download and the code snipped in a file called workflow.cwl and ran gives a plethora of errors regarding `Tool definition failed validation`

In short I could not follow through the tutorial because I was missing the input files, the linked CWL files did not seem to work out of the box, and the descriptions are not self contained. 

I admit to not having read the documentation at http://common-workflow-language.github.io/ I've tried but its to dense to read through and through, thus the importance of a good tutorial.

I hope this helped.
